### PR TITLE
Allow the use of the autoconfig client 

### PR DESCRIPTION
**Pull request by bolkedebruin**
This patch allows to use the AutoConfigClient of snakebite. This is important when using kerberos.

This is the first patch to fix this issue a follow up patch will also allow the manual config to do this, but snakebite needs to be updated first.

### CODE REVIEW

File: airflow/hooks/hdfs_hook.py [hunk 1]
```diff
@@ -32,19 +32,30 @@ def get_conn(self):
...
         client = None
-	''' When using HAClient, proxy_user must be the same, so is ok to always take the first '''
-	effective_user = self.proxy_user or connections[0].login
+
+        ''' When using HAClient, proxy_user must be the same, so is ok to always take the first '''
+        effective_user = self.proxy_user or connections[0].login
         if len(connections) == 1:
-            client = Client(connections[0].host, connections[0].port, use_sasl=use_sasl, effective_user=effective_user)
+            autoconfig = connections[0].extra_dejson.get('autoconfig', 'false')
+            if autoconfig == 'true':
```
Comment (mistercrunch) [inline airflow/hooks/hdfs_hook.py]: json supports booleans, so you may want to simply `if autoconfig:`, or more explicitely `if autoconfig is True or autoconfig.lower() == 'true` maybe

File: airflow/hooks/hdfs_hook.py [hunk 2]
```diff
@@ -32,19 +32,30 @@ def get_conn(self):
...
+                client = AutoConfigClient(effective_user=effective_user, use_sasl=use_sasl)
+            else:
+                client = Client(connections[0].host, connections[0].port,
+                                effective_user=effective_user, use_sasl=use_sasl)
+        elif len(connections) > 1:
```
Comment (mistercrunch) [inline airflow/hooks/hdfs_hook.py]: 52-54 are redundant with 55-57

Comment (mistercrunch) [inline airflow/hooks/hdfs_hook.py:52]: 52-54 are redundant with 55-57

**Comment by bolkedebruin** [comment]
Brrrr you are absolutely right. The horrors of merging :). Should be fixed now.

**Comment by bolkedebruin** [comment]
travis failure has nothing to do with the build

**Comment by mistercrunch** [comment]
yeah coveralls.io was choking this morning, I just restarted the build...

**Comment by mistercrunch** [comment]
:+1:

**Comment by garthcn** [comment]
hi @bolkedebruin , are you working on a patch to allow users to pass in `hdfs_namenode_principal` when they are not using auto-config? If not, I can probably submit a patch. 

Thanks for the awesome work :)

**Comment by bolkedebruin** [comment]
@garthcn It would be great if you could submit one. I dont have one ready right now. It is probably quite small anyway.

**Pull request merged**